library/src/main/java/com/novoda/downloadmanager/lib/BatchStatusRepository.java

```diff
@@ -75,6 +75,10 @@ int calculateBatchStatusFromDownloads(long batchId) {
             return DownloadStatus.RUNNING;
         }
 
+        if (statuses.hasOnlyCompleteAndSubmittedOrPendingStatuses()) {
```

Comment:
this method name is hard to parse because the operators AND and OR make it ambiguous.

Please help me understand this (I'm not too familiar with this library):

- `statuses` is a list of statuses for a single download set, a set of one or more files to download?
- this method determines whether that set of files is in the middle of downloading?

Does this mean that for a batch download, it will never return the PENDING state, even at the start?

Reply:
Yeah, I'm not a big fan of the name too, I tried to stay consistent with `hasOnlyCompleteAndSubmittedStatuses` but it doesn't work as well. I could rename it to something like `isBatchRunning` and maybe add a comment about what batch running means, if you think that's preferable?

Yes, `statuses` is the list status for each individual download file part of the batch. So the method determine if the current batch is currently downloading by checking that each individual file download is either COMPLETED or either PENDING or SUBMITTED. And it also need at least one file download with the COMPLETED status and one file with either the PENDING or SUBMITTED status.

So at the start of a batch download it will still return the PENDING status, since there will be no file download with the COMPLETED status in it.

Reply:
Will pick this up on Monday and try to get someone to weigh in also. Thanks for the quick reply and for the fix!